pennylane/templates/subroutines/qsvt.py

```diff
@@ -455,9 +455,17 @@ def compute_matrix(*args, **kwargs):
 
 def _qsp_to_qsvt(angles):
     r"""Converts qsp angles to qsvt angles."""
+    num_angles = len(angles)
     new_angles = qml.math.array(copy.copy(angles))
-    new_angles[0] += 3 * np.pi / 4
-    new_angles[-1] -= np.pi / 4
 
-    new_angles[1:-1] += np.pi / 2
+    for index in range(num_angles):
+        if index == 0:
+            update_val = 3 * np.pi / 4
+        elif index == num_angles - 1:
+            update_val = -np.pi / 4
+        else:
+            update_val = np.pi / 2
```

Comment:
The only thing I would say is that this for loop formulation seems sub-optimal in terms of performance. Are we able to use `set_index` on the range `[1:-1]`? I believe `jnp.array.at` should support multiple indices.

Reply:
I updated it, I ended up using `add_index` over `set_index` as we saved an extra array addition. Not sure if it's a massive performance gain, but better than nothing.